### PR TITLE
Fix #227 using lowercase Windows drive letters

### LINKED ISSUE #227: Not seeing test coverage overlay
### Environment

1. `node -v`: v8.9.1
2. `npm -v`: 5.5.1
3. `yarn -v`: 1.3.2
4. `npm ls react-scripts` (if you haven’t ejected): empty

1. Operating system: Windows 10

### Steps to Reproduce

Demo app: https://github.com/ngauthier/jest-vscode-bug-demo

1. Open VSCode
2. Open app/javascript/foo.jsx and test/javascript/foo.test.js and make changes and save

### Expected Behavior

See test coverage on my code.

### Actual Behavior

I don't see any test coverage on my code or my tests.

### Notes

Perhaps it's something to do with how I changed jests's directories?

Thank you for your time and this plugin, it is awesome!

### PR DESCRIPTION
**Pull request by seanpoulter**
As reported in #227, the code coverage overlay was not being shown on Windows when the Jest results had uppercase drive letters. This is caused by a standard in VS Code to use lowercase drive letters in URIs including the `document.fileName`.

This PR:
* groups files related to test results into one directory
* extracts the `TestResult` type into a new file
* adds utility methods to replace uppercase drive letters on Windows with the lowercase drive letter that's always provided by VS Code
* moves the "normalizing" the drive letters from the `TestResultProvider` into the `updateWithData` call
* tests the changes (except the call in `private updateWithData(...)`)

### CODE REVIEW

File: src/TestResults/TestResultProvider.ts [hunk 1]
```diff
@@ -1,29 +1,8 @@
...
-  start: Position
-  end: Position
-
-  status: TestReconciliationState
-  shortMessage?: string
-  terseMessage?: string
-
-  /** Zero-based line number */
-  lineNumberOfError?: number
-}
```
Comment (seanpoulter) [inline src/TestResults/TestResultProvider.ts]: This block has been moved into `TestResult.ts` above.

File: tests/TestResults/TestResultProvider.test.ts [hunk 2]
```diff
@@ -149,28 +149,24 @@ describe('TestResultProvider', () => {
...
       const sut = new TestResultProvider()
-      const results: any = {
-        testResults: [{ name: 'relative.js' }, { name: 'c:\\stays-lowercase' }, { name: 'D:\\changes-case' }],
-      }
-      sut.updateTestResults(results)
+      const results: any = {}
 
-      expect(updateFileWithJestStatus).toBeCalledWith({
-        testResults: [{ name: 'relative.js' }, { name: 'c:\\stays-lowercase' }, { name: 'd:\\changes-case' }],
-      })
```
Comment (seanpoulter) [inline tests/TestResults/TestResultProvider.test.ts]: The Jest results are now used as is in the `TestResultProvider`. The paths are changed in `updateWithData()`, and the method has been extracted into `TestResult.ts`.

**Comment by seanpoulter** [comment]
Sorry for all the spam!! Had a file change EOLs in the first commit and ended up diverging between local and remote. :disappointed:

**Comment by orta** [comment]
I missed this :D

**Comment by seanpoulter** [comment]
You didn't really miss -- I didn't assign it and figured you're busy too. 😉

**Comment by orta** [comment]
No worries, I think it's safe to assume I'll be reviewing - but if others are interested, don't let that stop you

**Pull request merged**